lighthouse/classes/event_properties/definitions.py

```diff
@@ -0,0 +1,213 @@
+        self.validate_param_not_missing(FIELD_EVENT_BARCODE)
+        self.validate_param_not_empty(FIELD_EVENT_BARCODE)
+        self.validate_param_no_whitespaces(FIELD_EVENT_BARCODE)
+        return self._validate
+
+    @cached_property
+    def value(self):
+        with self.retrieval_scope():
+            return self._params.get(FIELD_EVENT_BARCODE)
+
+    def add_to_warehouse_message(self, message):
+        for sample in self.value:
+            message.add_sample_as_subject(sample)
+
+
+class RunInfo(EventPropertyAbstract, ServiceCherrytrackMixin):
+    def __init__(self, run_id_property: RunID):
+        self.reset()
+        self.run_id_property = run_id_property
+
+    def validate(self):
+        return self.run_id_property.validate()
+
+    @property
+    def errors(self) -> List[str]:
+        self.validate()
+        return self._errors + self.run_id_property.errors  # type: ignore
+
+    @cached_property
+    def value(self):
+        with self.retrieval_scope():
+            return self.get_run_info(self.run_id_property.value)
+
+    def add_to_warehouse_message(self, message):
+        return None
+
+
+class PickedSamplesFromSource(EventPropertyAbstract, ServiceCherrytrackMixin, ServiceMongoMixin):
+    def __init__(self, barcode_property: PlateBarcode, run_id_property: RunID):
+        self.reset()
+        self.barcode_property = barcode_property
+        self.run_id_property = run_id_property
+
+    def validate(self):
+        return self.barcode_property.validate() and self.run_id_property.validate()
+
+    @property
+    def errors(self) -> List[str]:
+        self.validate()
+        return self._errors + self.barcode_property.errors + self.run_id_property.errors  # type: ignore
+
+    @cached_property
+    def value(self):
+        with self.retrieval_scope():
+            sample_uuids: List[str] = [  # type: ignore
+                sample[FIELD_CHERRYTRACK_LH_SAMPLE_UUID]
+                for sample in filter(
+                    lambda sample: sample[FIELD_EVENT_RUN_ID] == self.run_id_property.value,  # type: ignore
+                    filter(
+                        self.filter_pickable_samples,
+                        self.get_samples_from_source_plates(self.barcode_property.value),
+                    ),
+                )
+            ]
+            val: List[Dict[str, Any]] = list(self.get_samples_from_mongo(sample_uuids))
+            return val
+
+    def add_to_warehouse_message(self, message):
+        for sample in self.value:
+            message.add_sample_as_subject(sample)
+
+
+class UserID(EventPropertyAbstract, SimpleEventPropertyMixin):
+    def validate(self):
+        self.validate_param_not_missing(FIELD_EVENT_USER_ID)
+        self.validate_param_not_empty(FIELD_EVENT_USER_ID)
+        return self._validate
+
+    @cached_property
+    def value(self):
+        with self.retrieval_scope():
+            val = self._params.get(FIELD_EVENT_USER_ID)
+            if val is None:
```

Comment:
This seems to be doing the job of `validate()`?

Reply:
This is to make sure that if we don't have a value it will raise an exception. My idea was that value() is not allowed to return anything unless it is completely sure it is the right value, otherwise raise an exception.